### PR TITLE
SHL, SHR, SAR, and related tests

### LINKED ISSUE #350: SHL, SHR and SAR opcodes and correcponding Patch interface changes
For `v0.11` the least that needs to be done about `Constantinople` changes is changing the `Patch` interface as it's a breaking change that needs to be done __before__ the version bump.

### PR DESCRIPTION
**Pull request by mersinvald**
This PR adds related Constantinople opcodes.
Issue: #350 

Main question is how do we approach merging that?
Feature-by-feature or one big PR with all Constantinople changes included?
@whilei

Closes #350

### CODE REVIEW

File: src/patch/mod.rs [hunk 1]
```diff
@@ -177,6 +181,7 @@ impl Patch for EmbeddedByzantiumPatch {
     fn has_static_call() -> bool { true }
     fn has_revert() -> bool { true }
     fn has_return_data() -> bool { true }
+    fn has_bitwise_shift() -> bool { false }
```
Comment (whilei) [inline src/patch/mod.rs]: This Patch should be `EmbeddedConstantinoplePatch`, or?

Comment (mersinvald) [reply src/patch/mod.rs]: Oh. Dunno what I did, but my comment just got removed.

Patch name is OK here, but I gotta add an `EmbeddedConstantinoplePatch` indeed

**Comment by mersinvald** [comment]
Oops, forgot to update the `network` crates

File: src/patch/mod.rs (hunk 1 above)
Comment (mersinvald) [reply src/patch/mod.rs]: Actually decided to remove the `EmbeddedByzantiumPatch`: #353

**Review by whilei — APPROVED** [review]

**Review by r8d8 — CHANGES_REQUESTED** [review]
@mersinvald , can you please fix CI builds

**Comment by mersinvald** [comment]
@r8d8 working on it

**Review by r8d8 — APPROVED** [review]

**Pull request merged**